src/dns.rs

```diff
@@ -0,0 +1,58 @@
+mod local;
+
+use crate::types::Result;
+use std::net::IpAddr;
+use crate::dns::local::LocalOverrideTable;
+use crate::types;
+
+/// A DNS entry is a simple domain name to IP address mapping
+struct DNSEntry {
+    domain_name: String,
+    ip: IpAddr,
+}
+
+/// The main DNS resolver struct
+struct DnsResolver {
```

Comment:
This needs to be filled in later.. this pr just does the local resolver

Reply:
yes, this will be behind a trait

Reply:
The DNS resolver is now a trait.. I will have a few: the global dns resolver (that uses hickory dns), the local override table resolver and maybe others in the future. The dns components will consist of a global dns resolver, a local dns resolver, and a cache layer.  First, the resolver checks the cache layer. If time expired for the entry, or the entry is not found, the local dns resolver is queried. If there is nothing found, it will check the global dns resolver.